tests/testutils/src/main/kotlin/io/element/android/tests/testutils/WithFakeLifecycleOwner.kt

```diff
@@ -9,31 +9,54 @@ package io.element.android.tests.testutils
 
 import android.annotation.SuppressLint
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.InternalComposeApi
 import androidx.compose.runtime.Stable
+import androidx.compose.runtime.currentComposer
 import androidx.compose.runtime.getValue
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
 import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.LifecycleOwner
 import androidx.lifecycle.LifecycleRegistry
 import androidx.lifecycle.compose.LocalLifecycleOwner
+import app.cash.molecule.RecompositionMode
+import app.cash.molecule.moleculeFlow
+import app.cash.turbine.TurbineTestContext
+import app.cash.turbine.test
+import io.element.android.libraries.architecture.Presenter
 
+@OptIn(InternalComposeApi::class)
 @Stable
 @Composable
-fun <T> withFakeLifecycleOwner(lifecycleOwner: FakeLifecycleOwner = FakeLifecycleOwner(), block: @Composable () -> T): T {
-    var state: T? by remember { mutableStateOf(null) }
-    CompositionLocalProvider(LocalLifecycleOwner provides lifecycleOwner) {
-        state = block()
-    }
-    return state!!
+fun <T> withFakeLifecycleOwner(
+    lifecycleOwner: FakeLifecycleOwner = FakeLifecycleOwner(),
+    block: @Composable () -> T
+): T {
+    currentComposer.startProvider(LocalLifecycleOwner provides lifecycleOwner)
+    val state = block()
+    currentComposer.endProviders()
```

Comment:
```suggestion
    currentComposer.endProvider()
```

Seems to be more appropriate. `endProviders()` has to be used when `startProviders()` has been called. This is probably equivalent though.